command/src/config.rs

```diff
@@ -155,6 +154,55 @@ pub const DEFAULT_COMMAND_BUFFER_SIZE: usize = 1_000_000;
 /// maximum size of the buffer for the channels, in bytes. (2 MB)
 pub const DEFAULT_MAX_COMMAND_BUFFER_SIZE: usize = 2_000_000;
 
+#[derive(thiserror::Error, Debug)]
+pub enum ConfigError {
+    #[error("Could not parse socket address {address}: {error}")]
+    ParseSocketAddress { address: String, error: String },
+    #[error("env path not found: {0}")]
+    Env(String),
+    #[error("Could not open file {path_to_open}: {io_error}")]
+    FileOpen {
+        path_to_open: String,
+        io_error: String,
+    },
+    #[error("Could not read file {path_to_read}: {io_error}")]
+    FileRead {
+        path_to_read: String,
+        io_error: String,
+    },
+    #[error("the listener on {0} has incompatible options: it cannot use the expect proxy protocol and have a public_address field at the same time",)]
+    IncompatiblePublicAddress(String),
+    #[error("all the listeners for cluster {0} should have the same expect_proxy option")]
+    InconsistentExpectProxyOption(String),
+    #[error("Invalid '{0}' field for a TCP frontend")]
+    InvalidField(String),
+    #[error("invalid path {0:?}")]
+    InvalidPath(PathBuf),
+    #[error("listening address {0} is already used in the configuration")]
+    ListenerAddressAlreadyInUse(String),
+    #[error("Invalid '{0}' field to create a frontend")]
+    MissingField(String),
```

Comment:
Should be an enum with error `NotFound` with kind `Field(string)`, `Protocol` and `SavedState`